src/test/java/org/jabref/model/entry/AuthorListTest.java

```diff
@@ -31,6 +41,82 @@ public void testFixAuthorNatbib() {
                         .fixAuthorNatbib("John von Neumann and John Smith and Black Brown, Peter")));
     }
 
+    @Test
+    public void fixAuthorNatbibLatexFreeNullAuthorStringThrowsException() {
+        assertThrows(NullPointerException.class, () -> AuthorList.fixAuthorNatbibLatexFree(null));
+    }
+
+    @Test
+    public void fixAuthorNatbibLatexFreeEmptyAuthorStringForEmptyInput() {
+        assertEquals("", AuthorList.fixAuthorNatbibLatexFree(""));
+    }
+
+    @Test
+    public void fixAuthorNatbibLatexFreeCachesLatexFreeString() {
+        String cachedString = AuthorList
+                .fixAuthorNatbibLatexFree(MUHAMMAD_ALKHWARIZMI);
+        assertSame(cachedString, AuthorList
+                .fixAuthorNatbibLatexFree(MUHAMMAD_ALKHWARIZMI));
+    }
+
+    @Test
+    public void fixAuthorNatbibLatexFreeUnicodeOneAuthorNameFromLatex() {
+        assertEquals("al-Khwārizmī",
+                AuthorList.fixAuthorNatbibLatexFree(MUHAMMAD_ALKHWARIZMI));
+    }
+
+    @Test
+    public void fixAuthorNatbibLatexFreeUnicodeTwoAuthorNamesFromLatex() {
+        assertEquals("al-Khwārizmī and Böhm",
+                AuthorList.fixAuthorNatbibLatexFree(MUHAMMAD_ALKHWARIZMI
+                + " and " + CORRADO_BOHM));
+    }
+
+    @Test
+    public void fixAuthorNatbibLatexFreeUnicodeAuthorEtAlFromLatex() {
+        assertEquals("al-Khwārizmī et al.",
+                AuthorList.fixAuthorNatbibLatexFree(MUHAMMAD_ALKHWARIZMI
+                        + " and " + CORRADO_BOHM + " and " + KURT_GODEL));
+    }
+
+    @Test
+    public void fixAuthorNatbibLatexFreeUnicodeOneInsitutionNameFromLatex() {
+        assertEquals("The Banū Mūsā brothers",
+                AuthorList.fixAuthorNatbibLatexFree(BANU_MOSA));
+    }
+
+    @Test
+    public void fixAuthorNatbibLatexFreeUnicodeTwoInsitutionNameFromLatex() {
+        assertEquals("The Banū Mūsā brothers and The Banū Mūsā brothers",
+                AuthorList.fixAuthorNatbibLatexFree(BANU_MOSA
+                + " and " + BANU_MOSA));
+    }
+
+    @Test
+    public void fixAuthorNatbibLatexFreeUnicodeMixedAuthorsFromLatex() {
+        assertEquals("The Banū Mūsā brothers and Böhm",
+                AuthorList.fixAuthorNatbibLatexFree(BANU_MOSA
+                + " and " + CORRADO_BOHM));
+    }
+
+    @Test
+    public void fixAuthorNatbibLatexFreeOneInstitutionWithParanthesisAtStart() {
+        assertEquals("Łukasz Michał",
+                AuthorList.fixAuthorNatbibLatexFree("{{\\L{}}ukasz Micha\\l{}}"));
+    }
+
+    @Test
+    public void fixAuthorNatbibLatexFreeAuthorWithEscapedBrackets() {
+        assertEquals("Mic}h{ał",
+                AuthorList.fixAuthorNatbibLatexFree("{\\L{}}ukasz Mic\\}h\\{a\\l{}"));
+    }
+
+    @Test
+    public void fixAuthorNatbibLatexFreeInstituteAuthorWithEscapedBrackets() {
```

Comment:
Thanks for adding so many tests. This is really good. I would propose to split them a bit into two categories: tests for parsing and test for representation. The former should take a string and test against a `AuthorList`. The latter should take a `AuthorList` and test against a string. In this way, problems can be diagnosed more easily as it is clear in which part of the code the problem occurs. 

(I'm aware that the existing tests also often don't adhere to this splitting...that's the problem with legacy code :( )

Reply:
If I understand you correctly, all the tests I have added belongs to the latter category.

And, under the assumption that I do understand you correctly, I'll see if I can come up with some tests for the former (parsing) category.

Reply:
I think you can easily split the tests you added in these categories. For the parsing, simply parse the new strings `Mu{\\d{h}}ammad al-Khw{\\={a}}rizm{\\={i}}` etc you added and compare them to the hand-created AuthorList (say `MUHAMMAD_ALKHWARIZMI = new AuthorList(...)`%. Then the tests you already added can be converted to operate on `MUHAMMAD_ALKHWARIZMI ` (which is now an AuthorList and no longer the string).

Reply:
I have updated with an attempt at implementing this split. However, I am struggling a bit with creating test cases for parsing as I don't know enough about JabRef/BibTeX/BibLaTeX/Biber.

1. In JabRef an author's first name can contain `}` as in `Ku{\}}rt G{\"{o}}del` -> `Ku}rt Gödel`. This doesn't work in any combination* of BibTeX/BibLaTeX/Biber that I have tried.
2. The test case I am trying contains `{` as in `Ku{\{}rt G{\"{o}}del` -> `Ku{rt Gödel` which is not supported by JabRef nor any combination* of BibTeX/BibLaTeX/Biber.

So for now I am limiting the scope to not include escaped latex brackets. It is a very contrived case that is unlikely to happen, and since I don't understand why it doesn't work I probably shouldn't be writing test cases for it :P

With combinations of BibTeX/BibLaTeX/Biber, I mean that I tried BibTeX by itself, BibLaTeX with BibTeX backend, and BibLaTeX with biber backend.

Reply:
Sounds good to me. I think an escaped bracket in an author name is indeed very rare!